Automattic/woocommerce-payments
# Add onboarding flow state persistence

Fixes #5477 

#### Changes proposed in this Pull Request
- Add `get`, `set`, and `clear` onboarding flow state option methods in the Onboarding Service.
- Add editable `/onboarding/flow-state` REST route to update the option value from the JS client.
- Inject option value under `wcpaySettings.onboardingFlowState`.
- Update `<Stepper/>` to support `initialStep` prop.
- Persist state using `onStepChange` event, to ensure it's only persisted once per each step with valid data.
- Recover state using both `initialStep` and `intialData` only if `wcpaySettings.onboardingFlowState` exists.
- Clear the onboarding flow state on Stripe onboarding initialization, to avoid keeping an old state.
- Add a dismissable info banner, when there is a persisted state, to inform the merchant about it.

#### Testing instructions
> [!NOTE]
> To avoid the need to create new accounts you can enable `Force the WCPay plugin to act as disconnected from the WCPay Server` in WCPay Dev, causing the last onboarding step to just redirect back to the connect page.

- With PO enabled go to the onboarding flow.
- Pick live onboarding option.
- Fill in the personal details, and click on **Continue**.
- Refresh the page.
- You should land on the business step, just where you left it, and the new notice banner should be visible.
- Go back using the navbar button.
- Your previous details should be there.
- Finish the onboarding process.
- Start again, and confirm that the persisted state is gone.

#### Screenshot – New info banner
<img width="637" alt="Screenshot 2023-08-15 at 09 26 13" src="https://github.com/Automattic/woocommerce-payments/assets/7670276/1a43d9ce-bf8e-4d5a-9292-80956dddac66">

-------------------


**Post merge**
